src/meta/backup_engine.cpp

```diff
@@ -194,14 +194,36 @@ void backup_engine::backup_app_partition(const gpid &pid)
     _backup_status[pid.get_partition_index()] = backup_status::ALIVE;
 }
 
-void backup_engine::on_backup_reply(error_code err,
-                                    const backup_response &response,
-                                    gpid pid,
-                                    const rpc_address &primary)
+inline void backup_engine::handle_replica_backup_failed(const backup_response &response,
+                                                        const gpid pid)
 {
     dcheck_eq(response.pid, pid);
     dcheck_eq(response.backup_id, _cur_backup.backup_id);
 
+    derror_f("backup_id({}): backup for partition {} failed, response.err: {}",
+             _cur_backup.backup_id,
+             pid.to_string(),
+             response.err.to_string());
+    zauto_lock l(_lock);
+    // if one partition fail, the whole backup plan fail.
+    _is_backup_failed = true;
+    _backup_status[pid.get_partition_index()] = backup_status::FAILED;
+}
+
+inline void backup_engine::retry_backup(const dsn::gpid pid)
+{
+    tasking::enqueue(LPC_DEFAULT_CALLBACK,
+                     &_tracker,
+                     [this, pid]() { backup_app_partition(pid); },
+                     0,
+                     std::chrono::seconds(1));
```

Comment:
consider assign 10s

Reply:
Why

Reply:
it's not necessary to using too frequently value